test/common/tcp_proxy/upstream_test.cc

```diff
@@ -183,6 +183,17 @@ TYPED_TEST(HttpUpstreamTest, OnFailureCalledOnInvalidResponse) {
   this->upstream_->responseDecoder().decodeHeaders(std::move(headers), false);
 }
 
+TYPED_TEST(HttpUpstreamTest, PropagateResponseHeadersInCaseOfInvalidResponse) {
+  setupUpstream();
+  auto conn_pool_callbacks = std::make_unique<MockHttpConnPoolCallbacks>();
+  auto conn_pool_callbacks_raw = conn_pool_callbacks.get();
+  upstream_->setConnPoolCallbacks(std::move(conn_pool_callbacks));
+  EXPECT_CALL(*conn_pool_callbacks_raw, onFailure()).Times(0);
+  EXPECT_CALL(*conn_pool_callbacks_raw, onSuccess(_)).Times(0);
+  Http::ResponseHeaderMapPtr headers{new Http::TestResponseHeaderMapImpl{{":status", "418"}}};
+  upstream_->responseDecoder().decodeHeaders(std::move(headers), false);
+}
+
```

Comment:
Can we remove this test? I think the integration test is good enough.
Also , I see that it's exactly the same as  previous test "OnFailureCalledOnInvalidResponse".

Reply:
@0xquark ?

Reply:
Done!